pandas/io/parsers.py

```diff
@@ -1677,10 +1686,27 @@ def _infer_columns(self):
 
             columns = []
             for level, hr in enumerate(header):
-                line = self._buffered_line()
+                try:
+                    line = self._buffered_line()
 
-                while self.line_pos <= hr:
-                    line = self._next_line()
+                    while self.line_pos <= hr:
+                        line = self._next_line()
+
+                except StopIteration:
```

Comment:
why are you adding all of this code and making this very complicated?

the idea is to make things simpler. pls see if you can find a general soln.